src/index.html

```diff
@@ -11,7 +11,122 @@
     <link rel="stylesheet" href="./style.css">
   </head>
   <body>
-    <h1>HTML Form</h1>
     <script type="text/javascript" src="./main.js"></script>
+    <form>
+      <fieldset class="form__fieldset form-fieldset">
+        <legend class="form-fieldset__title">Personal information</legend>
+        <div class="field">
+          <label for="surname">Surname: </label>
+          <input type="text"
+            name="surname" 
+            required
+          >
+        </div>
+        <div class="field">
+          <label for="name">Name: </label>
+          <input type="text"
+            name="firstname" 
+            minlength="25"
+          >
+        </div>
+        <div class="field">
+          <label for="age">How old are you? </label>
+          <input type="number"
+            name="age"
+            min="0"
+            max="100"
+            value="12"
+          >
+        </div>
+        <div class="field">
+          <label for="birth">Full date of birth: </label>
+          <input type="date" name="birth">
+        </div>
+        <div class="field">
+          I accept the term of agreement 
+          <input type="checkbox" name="term">
+        </div>
+      </fieldset>
+      <fieldset class="form__fieldset">
+        <legend class="form-fieldset__title">Registration</legend>
+        <div class="field">
+          <label for="surname">E-mail: </label>
+          <input type="email"
+            name="email"
+            placeholder="email@example.com"
+          >
+        </div>
+        <div class="field">
+          <label for="name">Password: </label>
+          <input type="password"
+            name="password"
+            minlength="10"
+          >
+        </div>
+      </fieldset>
+      <fieldset class="form__fieldset form-fieldset">
+        <legend class="form-fieldset__title">An interesting fact about you!</legend>
+        <div class="field">
+          <label for="surname">Do you love cats? </label>
```

Comment:
it's not a surname, think this should be rather `cats`, names of the inputs below also should be changed :)